### ISSUE TITLE
when_multiline on csharp_prefer_braces not working 

### ISSUE THREAD

**Issue by MithrilMan** [dotnet/roslyn#40912]
**Version Used**: 
Visual Studio 16.4.2  Enterprise
**Steps to Reproduce**:
add in .editorconfig the line
`csharp_prefer_braces = when_multiline : error`

**Expected Behavior**:
shouldn't report the warning below and should work in the IDE
**Actual Behavior**:
![image](https://user-images.githubusercontent.com/5107375/72196650-3454fb00-341a-11ea-9ffe-0e582951000f.png)
![image](https://user-images.githubusercontent.com/5107375/72196717-8269fe80-341a-11ea-8c2a-a1c36101ae09.png)

**Comment by MithrilMan** [comment]
as a note, if this rule enforce just to put braces for expressions broken in multiple lines, I don't find this rule useful, we'd need a rule like `when_not_inline`

basically I'd like to use 
`if (clause) return/throw`
and I want CodeAnalysys require braces for
```
if(clause)
   return/throw
```
because I find always risky to put blocks without braces, so I only want to use them when I not inline them togheter like in the example.
Is it possible?

anyway the editorconfig is blaming so something doesn't work as expected to me.

**Comment by sharwell** [comment]
> shouldn't report the warning below

This is a duplicate of https://github.com/madskristensen/EditorConfigLanguage/issues/78.

> so I only want to use them when I not inline them togheter like in the example. Is it possible?

This is not currently supported, but you could write a custom analyzer for this. In addition to the support provided by the IDE analyzer, StyleCop Analyzers provides its own implementation of brace requirements. While neither of these supports the specific style you requested, you can use them as an example to create a new analyzer that handles your specific case.

**Comment by MithrilMan** [comment]
thanks for the link.
Do you think it's not a common scenario to have a rule for what I described?
May it be considered something to support officially?

**Comment by sharwell** [comment]
I haven't seen another request for the _specific combination_ you were looking for. The `when_multiline` is already quite close to what you are working on, as long as it's acceptable to not have a warning in one specific case.

**Comment by daniel-liuzzi** [comment]
I second @MithrilMan's opinion. This rule would be far more useful if it also enforced braces on anything that's not inline. The [documentation](https://docs.microsoft.com/en-us/visualstudio/ide/editorconfig-language-conventions?view=vs-2019#code-block-preferences) for `csharp_prefer_braces` state (emphasis mine):

- `true` - Prefer curly braces even **for one line of code**
- `false` - Prefer no curly braces if allowed
- `when_multiline` - Prefer curly braces **on multiple lines**

I was surprised to find `when_multiline` didn't complain about the code below and thought I had hit a bug:

```c#
if (true)
    Console.WriteLine("Hello World!");
```

I assume whether you consider the `if` a part of the statement or not (hence whether the code is considered "one line of code" or not) is all semantics, but the above code is "risky business" and there should be a rule to prevent it without enforcing braces everywhere.

**Comment by sharwell** [comment]
The `when_multiline` option was developed to support the [C# Coding Style](https://github.com/dotnet/runtime/blob/master/docs/coding-guidelines/coding-style.md) used for .NET Core and it's runtime libraries, along with most related open source projects (e.g. dotnet/roslyn). The recommended approach for other specific styles would be disabling this rule, and instead providing a custom analyzer for the specific desired code style.

**Comment by maxild** [comment]
It still seems like you have to use this (soon https://github.com/dotnet/roslyn/issues/43051#issuecomment-760512248) deprecated style to configure the option

```ini
csharp_prefer_braces = when_multiline:suggestion
dotnet_diagnostic.IDE0011.severity = warning
```

If using (the preferred syntax)

```ini
csharp_prefer_braces = when_multiline
dotnet_diagnostic.IDE0011.severity = warning
```

The `'Microsoft.CodeAnalysis.CSharp.Diagnostics.AddBraces.CSharpAddBracesDiagnosticAnalyzer` throws  `System.ArgumentNullException` with message 'Value cannot be null'.
